ucan-dns+web.md

```diff
@@ -0,0 +1,209 @@
+    }
+  ],
+  fct: [
+    // Evidence that "did:dns:w3.storage" has been delegated
+    // full authority over did:key:zAlice by the owner.
+    {
+      iss: "did:key:zAlice",
+      aud: "did:dns:w3.storage",
+      exp: null,
+      att: [
+        {
+          with: "did:key:zAlice",
+          can: "*"
+        }
+      ]
+    },
+  ]
+}
+```
+
+## Delegation of complete authority
+
+[UCAN][] specification does not describe an ability to delegate authority over the resources delegating prinipcal MAY hold in the future. This makes it impossible for `did:dns:w3.storage` key to delegate capability that `did:key:zAlice` will delegate to it in the future.
+
+To overcome this limitation here we propose delegation with `att: []` and `exp: null` to be treated as delegation of complete authority:
+
+```ts
+{
+  iss: "did:dns:w3.storage",
+  aud: "did:key:zService",
+  exp: null
+  att: []
+}
+```
+
+It can also be interpreted as `did:dns:w3.storage` stating to be "also known as"
+`did:key:zService` allowing it to delegate whatever `did:dns:w3.storage` CAN.
+
+## Pipelining dellegations
+
+[UCAN][] specification does not offer a way for two delegation chains to be pipelined into one. In our described scenario we have two delegation chains:
+
+1. `did:key:zAlice -> did:dns:w3.storage`
+2. `did:dns:w3.storage -> did:key:zService -> did:key:zRotation`
+
+From which we would like to construct a delegation of the capability delegated in (1) issued by the the (rightmost) audience in (2).
+
+```
+did:key:zAlice -> did:dns:w3.storage -> did:key:zService -> did:key:zRotation -> did:key:zAli
+```
+
+To accomplish this we propose that:
+
+- To issue delegation from `did:key:zRotation` with a valid proof chain (2) proving that `did:key:zRotation` has been delegated all capabilities from `did:dns:w3.storage`.
+- Embed proof chain (1) in `fct` as a verifable evidence that `did:dns:w3.storage` has been delegated capabilities for `did:key:zAlice`.
+
+```ts
+{
+  iss: "did:key:zRotation",
+  aud: "did:key:zAli",
+  exp: null,
+  att: [
+    {
+      with: "did:key:zAlice",
+      can: "*"
+    }
+  ],
+  prf: [
+    // Proof that did:key:zRotation has authority over
+    // did:dns:w3.storage delegated to it by did:key:zService
+    {
+      iss: "did:key:zService",
+      aud: "did:key:zRotation",
+      exp: null,
+      att: [],
+      // Proof that did:key:zService has been delegated
+      // authority over did:dns:w3.storage as long as
+      // did:key it resolves to is still the one that signed
+      prf: [{
+        iss: "did:dns:w3.storage",
+        aud: "did:key:zService",
+        exp: null
+        att: []
+      }]
+    }
+  ],
+  fct: [
+    // Evidence that "did:dns:w3.storage" has been delegated
+    // full authority over did:key:zAlice by the owner.
+    {
+      iss: "did:key:zAlice",
+      aud: "did:dns:w3.storage",
+      exp: null,
+      att: [
+        {
+          with: "did:key:zAlice",
+          can: "*"
+        }
+      ]
+    },
```

Comment:
_If_ we decide to support pipelining, I wonder if we can put this in the `prf` field and just support this directly?

Reply:
(Since this is "evidence", which is another name for a proof!)

Reply:
I think we could, in fact I proposed that as option (2) in https://github.com/ucan-wg/spec/issues/130. I mostly avoided here as it would directly violate principal alignment requirement